components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointUtil.java

```diff
@@ -140,7 +141,13 @@ public static final String getUserPortalUrl(String userPortalUrl) {
         if (StringUtils.isNotBlank(userPortalUrl)) {
             return userPortalUrl;
         }
-        return IdentityUtil.getServerURL(IdentityManagementEndpointConstants.USER_PORTAL_URL, true, true);
+        try {
+            return ServiceURLBuilder.create().addPath(IdentityManagementEndpointConstants.USER_PORTAL_URL).build()
+                    .getAbsolutePublicURL();
```

Comment:
At non tenated mode will this API return a tenanted URL.
User portal url was always tenanted I guess

Reply:
Or else has the tenant been appended some where else

Reply:
@ashensw , in legacy way when invoked from a tenant, has the user portal endpoint being tenant qualified appending tenant in path is some where else ?

If that's the case, that respective place will have to make sure that tenant is appended only in legacy mode right.
Because, the new API will tenant qualify the endpoint internally

Reply:
Nope. I don't see such a place that appends the tenant domain explicitly. This seems to used to get the old dashboard URL actually.

Reply:
ok. but I think we should further check how user portal worked previously in tenant cases.
Shall we track this and proceed with this fix